weatherman.py

```diff
@@ -0,0 +1,398 @@
+        except Exception:
+            pass
+        return []
+
+
+def get_month_number_from_string(string):
+    string = str(string).lower()
+    if string == "jan":
+        return 1
+    elif string == "feb":
+        return 2
+    elif string == "mar":
+        return 3
+    elif string == "apr":
+        return 4
+    elif string == "may":
+        return 5
+    elif string == "jun":
+        return 6
+    elif string == "jul":
+        return 7
+    elif string == "aug":
+        return 8
+    elif string == "sep":
+        return 9
+    elif string == "oct":
+        return 10
+    elif string == "nov":
+        return 11
+    elif string == "dec":
+        return 12
+    return None
+
+
+def get_month_string_from_number(number: int):
+    if number == 1:
+        return "Jan"
+    if number == 2:
+        return "Feb"
+    if number == 3:
+        return "Mar"
+    if number == 4:
+        return "Apr"
+    if number == 5:
+        return "May"
+    if number == 6:
+        return "Jun"
+    if number == 7:
+        return "Jul"
+    if number == 8:
+        return "Aug"
+    if number == 9:
+        return "Sep"
+    if number == 10:
+        return "Oct"
+    if number == 11:
+        return "Nov"
+    if number == 12:
+        return "Dec"
+    return ""
+
+
+def get_month_year_from_string(string):
+    month, year = string.split('/')
+    try:
+        month = int(month)
+        year = int(year)
+    except Exception:
+        raise "Invalid input"
+        return None, None
+    return month, year
+
+
+def average(lis):
+    sum_ = sum(lis)
+    return sum_ / len(lis)
+
+
+# Computes highest temperature, lowest temperature and humidity
+def minus_e(weather_data: WeatherData, year: int):
+    max_temp_list = []
+    min_temp_list = []
+    max_humidity_list = []
+    min_humidity_list = []
+
+    # Checking data of all 12 months
+    for i in range(0, 12):
+        month = i + 1
+        # Max temperature
+        col_name = weather_data.get_column_name_from_alias(
+                                "max_temperature")
+        if col_name is None:
+            print("Invalid column data")
+        else:
+            max_temp_column = weather_data.get_column_data(col_name,
+                                                           month, year)
+            if max_temp_column != []:
+                max_index = np.argmax(max_temp_column)
+                max_temp_list.append([max_temp_column[max_index],
+                                      max_index + 1, month])
+
+        # Min temperature
+        col_name = weather_data.get_column_name_from_alias(
+                                "min_temperature")
+        if col_name is None:
+            print("Invalid column alias")
+        else:
+            min_temp_column = weather_data.get_column_data(col_name,
+                                                           month, year)
+            if min_temp_column != []:
```

Comment:
move all this logic for min temperature to a separate method, and call that method here. Would make code more cleaner.